### PR TITLE
feat: support user config manifest compression

### PR DESCRIPTION
**Pull request by Taylor-lagrange**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Expose ManifestLogStorage compress configuration to user

Manifest log now supports compression but it is not enabled by default. I hope to expose the option to users whether to enable Manifest log compression.

expose a config `use_compress` to user whether to use Manifest log compression.

```toml
[storage.manifest]
checkpoint_on_startup = true
use_compress = true # false by default
```

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

Closes https://github.com/GreptimeTeam/greptimedb/issues/1394
Follow-up of https://github.com/GreptimeTeam/greptimedb/pull/1497

### CODE REVIEW

File: src/cmd/src/datanode.rs
Comment (evenyag) [inline src/cmd/src/datanode.rs:239]: nitpick: What about `compress`?
```suggestion
            compress = true
```

File: src/mito/src/config.rs
Comment (evenyag) [inline src/mito/src/config.rs:19]:
```suggestion
    pub compress_manifest: bool,
```

File: src/cmd/src/datanode.rs
Comment (Taylor-lagrange) [reply src/cmd/src/datanode.rs]: done

File: src/storage/src/region/tests.rs
Comment (killme2008) [inline src/storage/src/region/tests.rs:317]: Can we add a test for `compress=true` for region recovery? I think we can change this function into a normal function that accepts the boolean argument `compress`:

 ```rust
async fn test_recover_region_manifets(compress: bool) {
     ...
    let manifest = if compress {
           ...
    } else {
         ...
    };
     ....
}
```

And adds two tests:


```rust
#[tokio:test]
async fn test_recover_region_manifets_compress() {
    test_recover_region_manifets(true);
}

#[tokio:test]
async fn test_recover_region_manifets_uncompress() {
    test_recover_region_manifets(false);
}
```

**Comment by killme2008** [comment]
Maybe we need to add a new test suite for `compress=true` in sqlness tests. To ensure all is fine when enabling compression for region and table manifests.

https://github.com/GreptimeTeam/greptimedb/blob/develop/tests/conf/standalone-test.toml.template

It's not urgent. Can you create an issue to track it?

Thank you.

**Comment by Taylor-lagrange** [comment]
Consider add file `datanode-manifest-compress-test.toml.template` and `standalone-manifest-compress-test.toml.template` under path `/tests/conf`, then make it run and pass. Is any other needs to be consider? I have checked the code of sqlness runner quickly, only few line change will do.

**Comment by killme2008** [comment]
> Consider add file `datanode-manifest-compress-test.toml.template` and `standalone-manifest-compress-test.toml.template` under path `/tests/conf`, then make it run and pass. Is any other needs to be consider? I have checked the code of sqlness runner quickly, only few line change will do.

I think `standalone-manifest-compress-test.toml.template` is enough to cover. Thanks.

**Review by evenyag — APPROVED** [review]

**Comment by killme2008** [comment]
@Taylor-lagrange  Will you add the sqlness test suite or leave it to next PR?

**Review by killme2008 — APPROVED** [review]
LGTM

**Comment by Taylor-lagrange** [comment]
Sqlness code not complicate, maybe we can consider fork and make our own sqlness, The interface provided by the original code is not very flexible

**Comment by killme2008** [comment]
> Sqlness code not complicate, maybe we can consider fork and make our own sqlness, The interface provided by the original code is not very flexible

The sqlness maintainer is @waynexia , we can discuss this here or in slack channels.

**Pull request merged**
